src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs

```diff
@@ -234,5 +237,24 @@ internal static PackageSpec ToPackageSpec(ProjectNames projectNames, IEnumerable
 
             return packageSpec;
         }
+
+        private async Task EnsureInitializedAsync()
+        {
+            try
+            {
+                if (!_initialized)
+                {
+                    await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
This will be entered multiple times in a race - use AsyncLazyInitializer to prevent that.

Reply:
Yep, will handle that...I just wanted to confirm that this fixes some of the confirm.

All of our tests are currentl broken as well cause the AsyncProvider doesn't exist :)